server/channels/app/user.go

```diff
@@ -2327,6 +2525,9 @@ func (a *App) RestrictUsersSearchByPermissions(c request.CTX, userID string, opt
 	return options, nil
 }
 
+// UserCanSeeOtherUser checks if a user has permission to view another user's information.
```

Comment:
This is not necessarily a permission check but rather a check whether they are in a same team or channels.